src/profiler.h

```diff
@@ -134,6 +134,23 @@ class Profiler {
     void stopTimer();
     void timerLoop(void* timer_id);
 
+    void logIfNoSamples(Arguments& args, u64 excluded_samples) {
```

Comment:
I think the function is too big to be inlined in the header file. I'd move implementation to .cpp instead.